### PR TITLE
Try backporting to 4.14

### PR DESCRIPTION
**Pull request by chen2021-web**

### CODE REVIEW

File: kernel/arch.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #define __PT_SP_REG sp
 #define __PT_IP_REG pc
 
-#define PRCTL_SYMBOL "__arm64_sys_prctl"
```
Comment (tiann) [inline kernel/arch.h:18]: Add version check, such as:
```
#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 16, 0)
#define PRCTL_SYMBOL "__arm64_sys_prctl"
#else
#define PRCTL_SYMBOL "sys_prctl"
#endif

File: kernel/ksu.c [hunk 2]
```diff
@@ -175,7 +175,7 @@ extern void enable_sucompat();
 
 static int handler_pre(struct kprobe *p, struct pt_regs *regs)
 {
-	struct pt_regs *real_regs = (struct pt_regs *)PT_REGS_PARM1(regs);
```
Comment (tiann) [inline kernel/ksu.c:178]: Add version check

File: kernel/setup.sh [hunk 3]
```diff
@@ -16,7 +16,7 @@ else
      exit 127
 fi
 
-test -d "$GKI_ROOT/KernelSU" || git clone https://github.com/tiann/KernelSU
```
Comment (tiann) [inline kernel/setup.sh:19]: revert this change

File: kernel/sucompat.c [hunk 4]
```diff
@@ -236,7 +236,7 @@ static int read_handler_pre(struct kprobe *p, struct pt_regs *regs)
 }
 
 static struct kprobe faccessat_kp = {
-	.symbol_name = "do_faccessat",
+	.symbol_name = "sys_faccessat",
```
Comment (tiann) [inline kernel/sucompat.c:239]: Add version check

File: kernel/sucompat.c [hunk 5]
```diff
@@ -246,11 +246,7 @@ static struct kprobe newfstatat_kp = {
 };
 
 static struct kprobe execve_kp = {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
```
Comment (tiann) [inline kernel/sucompat.c:249]: Add version check

**Review by tiann — CHANGES_REQUESTED** [review]

**Comment by hmtheboy154** [comment]
It's better to make commits in English

**Comment by tiann** [comment]
> It's better to make commits in English

Don't worry, i will squash this commit when it get merged : )

File: kernel/sucompat.c (hunk 5 above)
Comment (chen2021-web) [reply kernel/sucompat.c:249]: I think the version check need change.Because in some version.(Such as 4.14)It use do_execveat_common

Comment (tiann) [reply kernel/sucompat.c:249]: how about?
4.19 - 5.9: __do_execve_file
5.9+: do_execveat_common
4.14 - 4.19: do_execveat_common

Comment (chen2021-web) [reply kernel/sucompat.c:249]: I think it's ok

File: kernel/arch.h (hunk 1 above)
Comment (tiann) [reply kernel/arch.h:18]: We should also modify it for x86_64

**Review by tiann — CHANGES_REQUESTED** [review]

**Review by tiann — APPROVED** [review]

**Comment by tiann** [comment]
Please don't touch this branch before it get merged.

**Review by Ylarod — APPROVED** [review]
